### ISSUE TITLE
built in plotting functions

### ISSUE THREAD

**Issue by kurokirasama** [nushell/nushell#5361]
### Related problem

It would be nice to have a nushell native in-terminal plotting tool, like gnuplot.

### Describe the solution you'd like

At first it could be to plot just numbers sequences, later, whatever that path led you

### Describe alternatives you've considered

gnuplot

**Comment by stormasm** [comment]
If you install the old version of nushell you could check it out...

https://github.com/nushell/nushell/releases/tag/0.44.0

**Comment by stormasm** [comment]
also if you have questions @onthebridgetonowhere may have some knowledge as awhile back he was considering porting over the commands....

**Comment by fdncred** [comment]
I agree that a new plotting functionality would be cool with line charts and bar charts and histograms, etc. I could never get the old charting commands to work right. They always gave me surprising results that didn't match excel or google sheets. but they'd be a good place to start looking at code.

**Comment by onthebridgetonowhere** [comment]
@stormasm thanks for the heads up. I have not had time to look at them in details, but who knows, maybe in the near future we can bring those back. 

@fdncred oh I'd be awesome to have proper plotting functionality. I am also thinking it'd be great if we could be able to plot to PDFs/HTMLs, using some external tools/apps (e.g., R, ggplot?)

**Comment by Jiehong** [comment]
One command exists, at least in 0.84.0: `histogram`, but that's it.

**Comment by fdncred** [comment]
There's also a nice `nu_plugin_plot` that you can find here https://github.com/nushell/awesome-nu#plugins that supports 3 type of terminal plots, iirc.

**Comment by cptpiepmatz** [comment]
I'd say this is out of scope of nushell itself. We even moved dataframes out of nushell itself, so it should be part of a plugin. Some plugins already exist that tackle this:
- [nu_plugin_plot](https://github.com/Euphrasiologist/nu_plugin_plot): A terminal plotting plugin for nushell.
- [nu_plugin_plotters](https://github.com/cptpiepmatz/nu-jupyter-kernel/tree/main/crates/nu_plugin_plotters): A nushell plugin for plotting charts.
- [nu_plugin_termplot](https://github.com/termplot/termplot): Beautiful plots in your terminal.

**Comment by wangjiawen2013** [comment]
Hi @kurokirasama   @fdncred  @cptpiepmatz 

I've been following the discussions around native terminal plotting and the push toward dedicated plugins in Nushell.

I’m currently developing [charton](https://github.com/wangjiawen2013/charton) — a pure Rust visualization engine inspired by the Grammar of Graphics. Unlike traditional ASCII plotters, charton supports:

Declarative spec & composable layers (Point, Line, Bar, Rect, Rule, Path, etc.)

Multi-panel Faceting (FacetWrap / FacetGrid) for multi-variable data exploration

High-density Unicode & ASCII rendering with clean layout management

Could you take a look at charton to see if its capabilities and architecture align with what the Nushell community is looking for? If this fits the vision for terminal data visualization in Nushell, I plan to wrap it into a native Nushell plugin (nu_plugin_charton) next month.

**Comment by fdncred** [comment]
looks cool to me. I'd like to see a nushell plugin with it. would love to see it work natively with nushell but also with the nushell polars plugin. nice work on this!

**Comment by kurokirasama** [comment]
> Hi [@kurokirasama](https://github.com/kurokirasama) [@fdncred](https://github.com/fdncred) [@cptpiepmatz](https://github.com/cptpiepmatz)
> 
> I've been following the discussions around native terminal plotting and the push toward dedicated plugins in Nushell.
> 
> I’m currently developing [charton](https://github.com/wangjiawen2013/charton) — a pure Rust visualization engine inspired by the Grammar of Graphics. Unlike traditional ASCII plotters, charton supports:
> 
> Declarative spec & composable layers (Point, Line, Bar, Rect, Rule, Path, etc.)
> 
> Multi-panel Faceting (FacetWrap / FacetGrid) for multi-variable data exploration
> 
> High-density Unicode & ASCII rendering with clean layout management
> 
> Could you take a look at charton to see if its capabilities and architecture align with what the Nushell community is looking for? If this fits the vision for terminal data visualization in Nushell, I plan to wrap it into a native Nushell plugin (nu_plugin_charton) next month.

Looks great to me!!, looking forward to seeing it in a plugin
